tests/runtests.py

```diff
@@ -190,6 +189,24 @@ def test_admin_login(self):
             )
             self.fail(message)
 
+    def test_device_monitoring_charts(self):
+        self.login()
+        self._wait_for_element()
+        self.get_resource('test-device', '/admin/config/device/')
+        self._wait_for_element()
+        self.base_driver.find_element(By.CSS_SELECTOR, 'ul.tabs li.charts').click()
```

Comment:
I tried adding only this test on the latest master and it fails here, I was expecting it to fail in the try/except block below, but that's not the case. Is this intended or not?

Failure output:

```
ERROR: test_device_monitoring_charts (__main__.TestServices)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/runtests.py", line 198, in test_device_monitoring_charts
    self.base_driver.find_element(By.CSS_SELECTOR, 'ul.tabs li.charts').click()
  File "/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py", line 748, in find_element
    return self.execute(Command.FIND_ELEMENT, {"using": by, "value": value})["value"]
  File "/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py", line 354, in execute
    self.error_handler.check_response(response)
  File "/lib/python3.8/site-packages/selenium/webdriver/remote/errorhandler.py", line 229, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {"method":"css selector","selector":"ul.tabs li.charts"}
  (Session info: chrome-headless-shell=127.0.6533.99); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
```

Reply:
I was not able to replicate this error locally. Did you also copy the changes in data.py and utis.py? 

Anyway, I have improved the test to make it more robust.